### ISSUE TITLE
My reconstructions are bad

### ISSUE THREAD

**Issue by jmandivarapu1** [bhpfelix/Variational-Autoencoder-PyTorch#4]
Hi,

I tried running your code but even after 40 to 50 epochs my reconstructions are like below. Can you help?

![Epoch_29_recon](https://user-images.githubusercontent.com/21245519/108228902-4751f880-710d-11eb-941c-fb56bafca7e0.jpg)

**Comment by seolhokim** [comment]
+1

**Comment by benearnthof** [comment]
I stumbled upon this codebase just now and reimplemented the main class here (to clear up questions about variable names and remove a lot of copy/paste in the layers.
[https://gist.github.com/benearnthof/6839a96a740548579a7a278cd035d622](https://gist.github.com/benearnthof/6839a96a740548579a7a278cd035d622) 
@jmandivarapu1 
To me it looks like the images were not normalized before training, I've added that to my CelebA dataset class and over the firest 5 training epochs the loss seems to steadily decrease so far. (Average loss during 5th Epoch: 1.22) I'll report back tomorrow with samples after 50+ epochs 

Edit: I noticed that we optimize Binary Cross Entropy for some reason which seemed not Ideal so I swapped it for MSEloss and now my reconstructions look like this: 
![image](https://github.com/bhpfelix/Variational-Autoencoder-PyTorch/assets/40601079/a8334d7d-51d6-4c6f-8506-03508b2becfb)
After 20 Epochs. Input vs Reconstructions. I'll update my gist tomorrow.
